includes/functions.php

```diff
@@ -0,0 +1,196 @@
+use Google\Web_Stories\Stories_Renderer\FieldState\CircleView;
+use Google\Web_Stories\Stories_Renderer\FieldState\ListView;
+
+/**
+ * Render stories based on the passed arguments.
+ *
+ * @param array $attrs Arguments for fetching stories.
+ * @param array $query_args Query arguments for stories.
+ *
+ * @return void
+ */
+function render_stories( $attrs = [], $query_args = [] ) {
+	$stories_obj = new Story_Query( $attrs, $query_args );
+	//phpcs:disable WordPress.Security.EscapeOutput.OutputNotEscaped
+	echo $stories_obj->render();
+}
+
+/**
+ * Returns list of stories based on the arguments passed to it.
+ *
+ * @param array $attrs Arguments for fetching stories.
+ * @param array $query_args Query arguments for stories.
+ *
+ * @return array
+ */
+function get_stories( $attrs = [], $query_args = [] ) {
+	$stories_obj = new Story_Query( $attrs, $query_args );
+
+	return $stories_obj->get_stories();
+}
+
+/**
+ * Render stories based on customizer settings.
+ *
+ * @return void
+ */
+function render_theme_stories() {
+	//phpcs:disable WordPress.Security.EscapeOutput.OutputNotEscaped
+	echo get_plugin_instance()->customizer->render_stories();
+}
+
+/**
+ * Returns field state for the provided view type.
+ *
+ * @param string $view View Type.
+ *
+ * @return GridView|ListView|CircleView|CarouselView
+ */
+function get_field( $view = 'grid' ) {
+
+	switch ( $view ) {
+		case 'grid':
+			$field_state = new GridView();
+			break;
+		case 'list':
+			$field_state = new ListView();
+			break;
+		case 'circles':
+			$field_state = new CircleView();
+			break;
+		case 'carousel':
+			$field_state = new CarouselView();
+			break;
+		default:
+			$default_field_state = new CircleView();
+			/**
+			 * Filters the fieldstate object.
+			 *
+			 * This depicts
+			 *
+			 * @since 1.3.0
+			 *
+			 * @param FieldState $default_field_state Field states for circle view.
+			 */
+			$field_state = apply_filters( 'web_stories_default_fieldstate', $default_field_state );
+	}
+
+	return $field_state;
+}
+
+/**
+ * Wrapper function for fetching field states
+ * based on the view types.
+ *
+ * Mainly uses FieldState and Fields classes.
+ *
+ * @SuppressWarnings(PHPMD.UnusedLocalVariable)
+ *
+ * @return array
+ */
+function fields_states() {
+	$views = get_layouts();
+
+	$fields = [
+		'title',
+		'author',
+		'date',
+		'image_align',
+		'excerpt',
+		'sharp_corners',
+		'archive_link',
+		'circle_size',
+	];
+
+	$field_states = [];
+
+	foreach ( $views as $view_type => $view_label ) {
+		$field_state = get_field( $view_type );
+		foreach ( $fields as $field ) {
+			$field_states[ $view_type ][ $field ] = [
+				'show'     => $field_state->$field()->show(),
+				'label'    => $field_state->$field()->label(),
+				'readonly' => $field_state->$field()->readonly(),
+			];
+		}
+	}
+
+	return $field_states;
+}
+
+/**
+ * Get supported layouts for web stories.
+ *
+ * @return mixed|void
+ */
+function get_layouts() {
```

Comment:
Don't see why this is a  separate function in this file. Should probably be moved elsewhere.

Reply:
Maybe this should be a trait, it is used in a number of places.